### ISSUE TITLE
Deprecate twisted.runner RPCServer and RPCServicesConf

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8123]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#8123|
|Type|enhancement|
|Created|2015-11-23 09:14:34Z|
|Branch|https://github.com/twisted/twisted/tree/runner-rpcserver-deprecation-8123-3|

From glyph

The parts that are totally broken, RPCServicesConf, RPCServer, and the code that instantiates it, are useful only to NFS server implementors, and don't work at all no matter your configuration.  So I would delete the whole implementation just so someone will get a clear notification in case they were importing one of these names but not actually using them, deprecate the module attributes, and remove them in the next available removal cycle.  So basically just leave an empty Service subclass there just as a courtesy (since that is slightly more polite, and only a tiny bit harder than just deleting it).

<details><summary>Searchable metadata</summary>

```
trac-id__8123 8123
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__None None
branch__branches_runner_rpcserver_deprecation_8123_3 branches/runner-rpcserver-deprecation-8123-3
branch_author__adiroiban__glyph adiroiban, glyph
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1448270074059031 1448270074059031
changetime__1484971285971000 1484971285971000
version__None None
owner__adiroiban adiroiban

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban commented|
|-|-|

While on this, maybe I can also update the module description


The purpose of inetdtap is to provide an 'inetd'-like server, to allow Twisted to invoke other programs to handle incoming sockets.  This is a useful thing as a "networking swiss army knife" tool, like netcat.  In fact, it does work, after a fashion; I wrote a file like this:

```
8123       stream  tcp  wait glyph  /bin/cat -
```

and then ran this:

put a blank line into 'rpc.conf', and then ran:

twistd -n inetd -f sampleinetd.conf -r rpc.conf 

and port 8123 properly became an echo server.

Now, in order to do this, you need a /etc/services file (this is hard-coded) that can be parsed by ServicesConf, which is _extremely_ picky, and probably the one that your OS comes with is still broken anyway.  But this code can work if properly configured, and those parts can be fixed.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban commented|
|-|-|

(In [46412]) Branching to runner-rpcserver-deprecation-8123.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban removed owner|
|-|-|

I push the initial changes.

Test fail as I think that I need to work on improving the @deprecated decorator... rather than updating the way @deprecated is called in this branch.

I have also tried to document the `twistd -n inetd` command, but I could not find a way to inject more help info into `twistd -n inetd --help` command.

Please advise how to have the inetd usage info in the command line help.

I have manually tested the inetd and it works OK for echo server.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

This is blocked on [#8124](https://github.com/twisted/twisted/issues/8124), it seems?

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph set owner to @adiroiban|
|-|-|

This is failing tests (perhaps due to that dependency?) so pulling it back out of review for now.  Otherwise it looks fine; hope we can land once the dependency is resolved.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph commented|
|-|-|

(In [47068]) Branching to runner-rpcserver-deprecation-8123-2.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_7"></a>|@glyph commented|
|-|-|

I merged forward, but the tests are still failing.  [#8124](https://github.com/twisted/twisted/issues/8124) is now resolved though, so perhaps this could be fixed now.  Adi?

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_8"></a>|@adiroiban commented|
|-|-|

The branch is using the wrong deprecation helpers.

Also the advertised version is wrong.

I have pushed a change but a new commit is required for updating the tests.

I am not sure if I am using the right deprecation methods. Please review [#8082](https://github.com/twisted/twisted/issues/8082)

something is very broken on my system ... and is a bit late here

```
(py27-tests)$ ./bin/trial twisted/runner/inetdconf.py

-------------------------------------------------------------------------------

PASSED
```

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_9"></a>|@adiroiban removed owner|
|-|-|

OK. Stupid error due to trying to program while tired :( .. I was not running the test files.

--------

Test updated. Please review first the documentation talking about deprecation conventions.

This branch is ready for a final review.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_10"></a>|@glyph set owner to @adiroiban|
|-|-|

Thanks for the update, Adi.

1. it's 16.2.0 now :-\
2. There are still failures on the buildbot; the one on Windows is real, although the others may be spurious.
3. The feature doesn't actually work, so preserving the command-line flag doesn't really have any value.  That explicitly isn't a supported compatibility surface in CompatibilityPolicy; you can just delete the entry in `optParameters` without deprecating first.

Otherwise it looks pretty good.  However, given that the test failures seem to be persistent on this change I want to be extra careful, so please resubmit for review when these points are addressed and the builders are green.  Hopefully the final review can be quick :).

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_11"></a>|@adiroiban commented|
|-|-|

(In [47282]) Branching to runner-rpcserver-deprecation-8123-3.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_12"></a>|@adiroiban removed owner|
|-|-|

I think that the builder should be now green.

please review.

thanks!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_13"></a>|@glyph set owner to @adiroiban|
|-|-|

Please fix the one outstanding twistedchecker error (I think by adding literally a single space) and then land :).

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_14"></a>|@adiroiban commented|
|-|-|

twistedchecker is for this line and I think that this is a false positive.

```
    commentChar = '#'
```

will merge the branch as it is.

thanks for the review

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_15"></a>|@adiroiban set status to `closed`|
|-|-|

(In [47323]) Merge runner-rpcserver-deprecation-8123-3: Deprecate twisted.runner RPCServer and RPCServicesConf.

Author: adiroiban
Reviewer: glyph
Fixes: [#8123](https://github.com/twisted/twisted/issues/8123)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_16"></a>|@glyph commented|
|-|-|

Adding some Cc:'s, and:

How much longer until this (particularly, `portmap.c`) can die in a removal, now that it's deprecated?
